cprover_bindings/src/goto_program/typ.rs

```diff
@@ -570,6 +606,41 @@ impl Type {
         }
     }
 
+    /// A transparent type wraps a base type inside a struct, and has the same in-memory layout.
+    /// For example,
+    /// ```rust
+    ///     struct TransparentWrapper { int x };
+    /// ```
+    /// We define it recursively as a type which has exactly one field, which it itself either
+    /// a transparent type, or is a scalar type.
+    pub fn is_transparent_type(&self, st: &SymbolTable) -> bool {
+        (self.is_struct_like() || self.is_union_like())
+            && self.unwrap_transparent_type(st).is_some()
+    }
+
+    /// Given a transparent type (see comment on `Type::is_transparent_type()`),
```

Comment:
nit: Can you please update this to something like: "If type is transparent, extract the type it wraps"? The current documentation makes it sound like there is a pre-condition that `self` must be transparent.

Reply:
done